oap-server/server-library/library-util/src/test/java/org/apache/skywalking/oap/server/library/util/FileChangeMonitorTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.library.util;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileChangeMonitorTest {
+    private static String FILE_NAME = "FileChangeMonitorTest.tmp";
```

Comment:
better place it else where, in case the RAT plugin scans it occasionally

Reply:
What is `FileSystems.getDefault().newWatchService()` for? That doesn't look like a watcher with callback to me. Do I miss anything?

Reply:
I think I clear the file in the end of test case. Where should I put? It is different in different OS systems, with different access control maybe.

Reply:
Another concern is noticing this
```
     * @throws  UnsupportedOperationException
     *          If this {@code FileSystem} does not support watching file system
     *          objects for changes and events. This exception is not thrown
     *          by {@code FileSystems} created by the default provider.
```
Seems like we need to count on the JRE implementation somehow. I don't know the black magic inside now :)

Reply:
> What is `FileSystems.getDefault().newWatchService()` for? That doesn't look like a watcher with callback to me. Do I miss anything?

Hope the code snippets helps, my concern is that the built-in tool is well optimized and tested, and reduce our maintenance work:

```java
    public static void main(String[] args) throws IOException, InterruptedException {
        final WatchService watchService = FileSystems.getDefault().newWatchService();
        final Path directoryToWatch = Paths.get("/tmp/directory-to-watch");
        directoryToWatch.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);

        for (final WatchKey readyKey = watchService.take(); readyKey.isValid(); readyKey.reset()) {
            for (final WatchEvent<?> event : readyKey.pollEvents()) {
                assert event.kind().equals(StandardWatchEventKinds.ENTRY_MODIFY);
                final Path path = (Path) event.context();
                System.out.println("file is modified = " + directoryToWatch.resolve(path));
            }
        }
    }
```

Reply:
My question is, when the exception happens, I never used this API before. Evaluating the real case.

Reply:
This exception is rarely to see in the JDK classes. They only say this when necessary. I don't what this simple feature crashes.

Reply:
I agree with @kezhenxu94 . JDK builtin WatchSerivce is a better choice for the file modified notify service.